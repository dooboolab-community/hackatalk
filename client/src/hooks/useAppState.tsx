import {AppState, AppStateStatus} from 'react-native';
import {useEffect, useState} from 'react';

export default function useAppState(): AppStateStatus {
  const currentState = AppState.currentState;
  const [appState, setAppState] = useState(currentState);

  function onChange(newState: AppStateStatus): void {
    setAppState(newState);
  }

  useEffect(() => {
    AppState.addEventListener('change', onChange);

    return (): void => {
      AppState.removeEventListener('change', onChange);
    };
  });

  return appState;
}
