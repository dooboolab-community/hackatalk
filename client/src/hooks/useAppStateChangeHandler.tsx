import {AppState, AppStateStatus} from 'react-native';

import {useEffect} from 'react';

type AppStateChangeHandler = (state: AppStateStatus) => void;

/**
 * Custom hook for adding app state change event handler.
 * @param handler Callback function to handle app state change.
 */
export default function useAppStateChangeHandler(
  handler: AppStateChangeHandler,
): void {
  useEffect(() => {
    AppState.addEventListener('change', handler);

    return (): void => {
      AppState.removeEventListener('change', handler);
    };
  }, [handler]);
}
