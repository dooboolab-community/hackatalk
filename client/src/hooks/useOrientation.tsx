import * as ScreenOrientation from 'expo-screen-orientation';

import {useEffect, useState} from 'react';

import {Dimensions} from 'react-native';

export enum Orientation {
  PORTRAIT = 'portrait',
  LANDSCAPE = 'landscape',
}

export default function useAppState(): Orientation {
  const isPortrait =
    Dimensions.get('window').width < Dimensions.get('window').height;

  const [orientation, setOrientation] = useState<Orientation>(
    isPortrait ? Orientation.PORTRAIT : Orientation.LANDSCAPE,
  );

  useEffect(() => {
    const subscription = ScreenOrientation.addOrientationChangeListener(
      ({orientationInfo}) => {
        const isNewOrientationPortrait =
          orientationInfo.orientation ===
            ScreenOrientation.Orientation.PORTRAIT_UP ||
          orientationInfo.orientation ===
            ScreenOrientation.Orientation.PORTRAIT_DOWN;

        setOrientation(
          isNewOrientationPortrait
            ? Orientation.PORTRAIT
            : Orientation.LANDSCAPE,
        );
      },
    );

    return (): void => {
      ScreenOrientation.removeOrientationChangeListener(subscription);
    };
  });

  return orientation;
}
