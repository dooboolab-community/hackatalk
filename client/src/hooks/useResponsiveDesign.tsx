import '@expo/match-media'; // For polyfill side-effect.

import {useMediaQuery} from 'react-responsive';

export type Media = 'mobile' | 'tablet' | 'desktop';

/**
 * React hook that returns size category for responsive design.
 */
function useResponsiveDesign(): Media {
  const isMobile = useMediaQuery({
    maxWidth: 480,
  });

  const isTablet = useMediaQuery({
    maxWidth: 768,
  });

  if (isMobile) return 'mobile';

  if (isTablet) return 'tablet';

  return 'desktop';
}

export default useResponsiveDesign;
