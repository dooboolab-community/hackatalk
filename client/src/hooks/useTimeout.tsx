import React from 'react';

const useTimeout = (callback: () => void, delay: number): void => {
  const savedCallback = React.useRef<() => void>();

  React.useEffect(() => {
    savedCallback.current = callback;
  }, [callback]);

  React.useEffect(() => {
    function tick(): void {
      savedCallback.current?.();
    }

    if (delay !== null) {
      let id = setTimeout(tick, delay);

      return () => clearTimeout(id);
    }
  }, [delay]);
};

export default useTimeout;
