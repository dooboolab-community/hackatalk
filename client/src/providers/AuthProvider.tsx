import React, {useReducer} from 'react';

import {User} from '../types/graphql';
import createCtx from '../utils/createCtx';

export interface State {
  user: User | undefined;
}

interface Context {
  state: State;
  setUser(user: User | undefined): void;
}

const [useCtx, Provider] = createCtx<Context>();

// eslint-disable-next-line no-shadow
export enum ActionType {
  SetUser = 'set-user',
}

type SetUserAction = {
  type: ActionType.SetUser;
  payload: {user: User};
};
type Action = SetUserAction;

interface Props {
  children?: React.ReactElement;
  initialAuthUser?: User;
}

type Reducer = (state: State, action: Action) => State;

const setUser = (dispatch: React.Dispatch<SetUserAction>) => (
  authUser: User,
): void => {
  dispatch({
    type: ActionType.SetUser,
    payload: {user: authUser},
  });
};

const initialState: State = {
  user: undefined,
};

// eslint-disable-next-line default-param-last
const reducer: Reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.SetUser:
      return {...action.payload};
    default:
      return state;
  }
};

function AuthProvider(props: Props): React.ReactElement {
  const initialAuthState = {
    user: props.initialAuthUser,
  };

  const [state, dispatch] = useReducer<Reducer>(reducer, initialAuthState);

  const actions = {
    setUser: setUser(dispatch),
  };

  return <Provider value={{state, ...actions}}>{props.children}</Provider>;
}

const AuthContext = {
  useAuthContext: useCtx,
  AuthProvider,
};

export {useCtx as useAuthContext, AuthProvider};
export default AuthContext;
