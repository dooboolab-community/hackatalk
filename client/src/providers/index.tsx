export * from './AuthProvider';
export * from './DeviceProvider';
export * from './ProfileModalProvider';
