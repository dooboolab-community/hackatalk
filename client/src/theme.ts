import type {DoobooTheme} from 'dooboo-ui';

/* eslint-disable */
export const colors = {
  brand: '#1E6EFA',
  paleBlue05: '#E2F2FF',
  paleBlue10: '#8BC9FF',
  green10: '#A1EEDB',
  green30: '#00D4AB',
  green50: '#28DB98',
  gray05: '#F3F9FF',
  success: '33ff2f',
  danger: '#ff002e',
  warning: '#f2df0f',
  info: '#3a74e7',
  light: '#ededed',
  dark: '#151C22',
  messagePrimary: '#8BC9FF',
  messageSecondary: '#E2F2FF',
  apple: '#151E22',
  google: '#E04238',
  facebook: '#345997',
};
/* eslint-enable */

export const light = {
  background: 'white',
  paper: '#F2F5F6',
  disabled: '#C4C4C4',
  line: '#gray05',
  text: '#000',
  textLight: '#1B1B1B',
  textContrast: 'white',
  placeholder: '#6D6D6D',
  header: colors.brand,
  card: 'white',
  active: colors.brand,
  primary: colors.brand,
  primaryLight: colors.paleBlue10,
  secondary: colors.green10,
  button: colors.brand,
};

export type Theme = typeof light & DoobooTheme;

export const dark = {
  background: '#000',
  paper: colors.dark,
  disabled: '#515151',
  line: '#000',
  text: 'white',
  textLight: '#D3D3D3',
  textContrast: '#000',
  placeholder: '#6D6D6D',
  header: '#000',
  card: colors.dark,
  active: colors.paleBlue10,
  primary: '#fff',
  primaryLight: '#151A25',
  secondary: colors.green30,
  button: colors.paleBlue10,
};
