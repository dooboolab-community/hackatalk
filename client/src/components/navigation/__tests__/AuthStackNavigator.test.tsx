import 'react-native';

import { AllProviders } from '../../../providers';
import { AuthProvider } from '../../../providers/AuthProvider';
import { ThemeProvider as DoobooThemeProvider } from '@dooboo-ui/theme';
import { NavigationContainer } from '@react-navigation/native';
import React from 'react';
import { RelayEnvironmentProvider } from 'react-relay/hooks';
import StackNavigator from '../AuthStackNavigator';
import { createMockEnvironment } from 'relay-test-utils';
import {
  render,
} from '@testing-library/react-native';

const component = (
  <AllProviders>
    <NavigationContainer>
      <StackNavigator />
    </NavigationContainer>
  </AllProviders>
);

describe('[Stack] navigator', () => {
  it('should renders without crashing', async () => {
    const json = render(component).toJSON();

    expect(json).toBeTruthy();
    expect(json).toMatchSnapshot();
  });
});
