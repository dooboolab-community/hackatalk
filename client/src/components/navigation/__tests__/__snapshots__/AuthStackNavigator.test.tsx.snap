// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Stack] navigator should renders without crashing 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "opacity": 1,
            }
          }
        />
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onGestureEnd={[Function]}
          onOpen={[Function]}
          onTransitionStart={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
                "marginTop": 0,
              }
            }
          >
            <View
              collapsable={false}
              forwardedRef={[Function]}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              style={
                Object {
                  "flex": 1,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "backgroundColor": "rgb(242, 242, 242)",
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": -1,
                    },
                    "shadowOpacity": 0.3,
                    "shadowRadius": 5,
                    "top": 0,
                    "width": 3,
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "overflow": "hidden",
                    },
                    Array [
                      Object {
                        "backgroundColor": "rgb(242, 242, 242)",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "column-reverse",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <RCTSafeAreaView
                      emulateUnlessSupported={true}
                      style={
                        Array [
                          Object {
                            "backgroundColor": "#ffffff",
                            "flexBasis": 0,
                            "flexGrow": 1,
                            "flexShrink": 1,
                            "justifyContent": "center",
                          },
                        ]
                      }
                    >
                      <RCTScrollView
                        style={
                          Array [
                            Object {
                              "alignSelf": "center",
                              "width": "100%",
                            },
                          ]
                        }
                      >
                        <View>
                          <View
                            accessible={true}
                            focusable={true}
                            onClick={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Object {
                                "left": Object {
                                  " __value": 0,
                                },
                                "opacity": 1,
                                "position": "absolute",
                                "top": Object {
                                  " __value": 0,
                                },
                                "transform": Array [
                                  Object {
                                    "scale": AnimatedValue {
                                      " __value": 0,
                                    },
                                  },
                                ],
                                "zIndex": 15,
                              }
                            }
                            testID="theme-test"
                          >
                            <Image
                              source={
                                Object {
                                  "canInstrument": true,
                                  "createTransformer": [Function],
                                  "getCacheKey": [Function],
                                  "process": [Function],
                                }
                              }
                              style={
                                Object {
                                  "height": 80,
                                  "resizeMode": "cover",
                                  "width": 80,
                                }
                              }
                            />
                          </View>
                          <View
                            style={
                              Array [
                                Object {
                                  "marginBottom": 40,
                                  "marginLeft": 40,
                                  "marginRight": 40,
                                  "marginTop": 40,
                                },
                              ]
                            }
                          >
                            <View
                              style={
                                Array [
                                  Object {
                                    "marginBottom": 72,
                                    "marginTop": 44,
                                  },
                                ]
                              }
                            >
                              <View
                                style={
                                  Object {
                                    "height": 72,
                                  }
                                }
                              />
                              <Text
                                style={
                                  Array [
                                    Object {
                                      "alignSelf": "flex-start",
                                      "color": "#151A25",
                                      "fontSize": 20,
                                      "fontWeight": "bold",
                                      "marginLeft": 6,
                                    },
                                  ]
                                }
                              >
                                Hello
                              </Text>
                            </View>
                            <View
                              style={
                                Array [
                                  Object {
                                    "alignSelf": "stretch",
                                    "flexDirection": "column",
                                  },
                                  Object {
                                    "marginBottom": 20,
                                  },
                                ]
                              }
                            >
                              <View
                                style={
                                  Array [
                                    Object {
                                      "alignItems": "center",
                                      "flexDirection": "row",
                                      "justifyContent": "space-between",
                                      "width": "100%",
                                    },
                                    Object {
                                      "borderBottomWidth": 0.6,
                                      "borderColor": "#2E2E2E",
                                    },
                                    undefined,
                                    false,
                                    undefined,
                                  ]
                                }
                              >
                                <Text
                                  style={
                                    Array [
                                      Object {
                                        "color": "#b9b9c4",
                                        "fontSize": 14,
                                        "fontWeight": "bold",
                                      },
                                      undefined,
                                      false,
                                      Object {
                                        "width": 110,
                                      },
                                    ]
                                  }
                                >
                                  Email
                                </Text>
                                <TextInput
                                  allowFontScaling={true}
                                  autoCapitalize="none"
                                  multiline={false}
                                  onBlur={[Function]}
                                  onChangeText={[Function]}
                                  onFocus={[Function]}
                                  onSubmitEditing={[Function]}
                                  placeholder="hello@example.com"
                                  placeholderTextColor="#676c7a"
                                  rejectResponderTermination={true}
                                  style={
                                    Array [
                                      Object {
                                        "color": "#2c374e",
                                        "flexBasis": 0,
                                        "flexGrow": 1,
                                        "flexShrink": 1,
                                        "fontSize": 14,
                                        "fontWeight": "bold",
                                        "height": "100%",
                                        "paddingBottom": 16,
                                        "paddingTop": 16,
                                      },
                                      Object {
                                        "color": "#151A25",
                                      },
                                      Object {
                                        "textAlign": "left",
                                      },
                                    ]
                                  }
                                  testID="input-email"
                                  underlineColorAndroid="transparent"
                                  value=""
                                />
                              </View>
                            </View>
                            <View
                              style={
                                Array [
                                  Object {
                                    "alignSelf": "stretch",
                                    "flexDirection": "column",
                                  },
                                  Object {
                                    "marginBottom": 20,
                                  },
                                ]
                              }
                            >
                              <View
                                style={
                                  Array [
                                    Object {
                                      "alignItems": "center",
                                      "flexDirection": "row",
                                      "justifyContent": "space-between",
                                      "width": "100%",
                                    },
                                    Object {
                                      "borderBottomWidth": 0.6,
                                      "borderColor": "#2E2E2E",
                                    },
                                    undefined,
                                    false,
                                    undefined,
                                  ]
                                }
                              >
                                <Text
                                  style={
                                    Array [
                                      Object {
                                        "color": "#b9b9c4",
                                        "fontSize": 14,
                                        "fontWeight": "bold",
                                      },
                                      undefined,
                                      false,
                                      Object {
                                        "width": 110,
                                      },
                                    ]
                                  }
                                >
                                  Password
                                </Text>
                                <TextInput
                                  allowFontScaling={true}
                                  autoCapitalize="none"
                                  multiline={false}
                                  onBlur={[Function]}
                                  onChangeText={[Function]}
                                  onFocus={[Function]}
                                  onSubmitEditing={[Function]}
                                  placeholder="******"
                                  placeholderTextColor="#676c7a"
                                  rejectResponderTermination={true}
                                  secureTextEntry={true}
                                  style={
                                    Array [
                                      Object {
                                        "color": "#2c374e",
                                        "flexBasis": 0,
                                        "flexGrow": 1,
                                        "flexShrink": 1,
                                        "fontSize": 14,
                                        "fontWeight": "bold",
                                        "height": "100%",
                                        "paddingBottom": 16,
                                        "paddingTop": 16,
                                      },
                                      Object {
                                        "color": "#151A25",
                                      },
                                      Object {
                                        "textAlign": "left",
                                      },
                                    ]
                                  }
                                  testID="input-password"
                                  underlineColorAndroid="transparent"
                                  value=""
                                />
                              </View>
                            </View>
                            <View
                              style={
                                Array [
                                  Object {
                                    "flexDirection": "row",
                                    "marginTop": 12,
                                    "width": "100%",
                                  },
                                ]
                              }
                            >
                              <View
                                accessible={true}
                                focusable={true}
                                onClick={[Function]}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                style={
                                  Object {
                                    "backgroundColor": "#ffffff",
                                    "borderColor": "#1E6EFA",
                                    "borderRadius": 0,
                                    "borderWidth": 1,
                                    "opacity": 1,
                                    "width": "100%",
                                  }
                                }
                                testID="btn-sign-up"
                              />
                              <View
                                style={
                                  Object {
                                    "width": 12,
                                  }
                                }
                              />
                              <View
                                accessible={true}
                                focusable={true}
                                onClick={[Function]}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                style={
                                  Object {
                                    "backgroundColor": "#1E6EFA",
                                    "borderRadius": 0,
                                    "opacity": 1,
                                    "width": "100%",
                                  }
                                }
                                testID="btn-sign-in"
                              />
                            </View>
                            <View
                              accessible={true}
                              focusable={true}
                              onClick={[Function]}
                              onResponderGrant={[Function]}
                              onResponderMove={[Function]}
                              onResponderRelease={[Function]}
                              onResponderTerminate={[Function]}
                              onResponderTerminationRequest={[Function]}
                              onStartShouldSetResponder={[Function]}
                              style={
                                Object {
                                  "alignSelf": "center",
                                  "marginBottom": 44,
                                  "opacity": 1,
                                  "paddingBottom": 20,
                                  "paddingLeft": 20,
                                  "paddingRight": 20,
                                  "paddingTop": 20,
                                }
                              }
                              testID="btn-find-pw"
                            >
                              <Text
                                style={
                                  Array [
                                    Object {
                                      "color": "#1E6EFA",
                                      "textDecorationLine": "underline",
                                    },
                                  ]
                                }
                              >
                                Forgot password?
                              </Text>
                            </View>
                            <View
                              style={
                                Array [
                                  Object {
                                    "marginBottom": 24,
                                  },
                                ]
                              }
                            >
                              <View
                                accessible={true}
                                focusable={true}
                                onClick={[Function]}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                style={
                                  Object {
                                    "backgroundColor": "#ffffff",
                                    "borderColor": "#151E22",
                                    "borderRadius": 100,
                                    "borderWidth": 1,
                                    "height": 48,
                                    "marginBottom": 12,
                                    "opacity": 1,
                                    "width": "100%",
                                  }
                                }
                                testID="btn-apple"
                              />
                              <test
                                clientId="xxxx"
                                clientSecret="xxxx"
                                onUserCreated={[Function]}
                                socialProvider="facebook"
                                svgIcon={
                                  <SvgMock
                                    fill="#ffffff"
                                    height={18}
                                    width={18}
                                  />
                                }
                              />
                              <test
                                clientId="xxxx"
                                clientSecret="xxxx"
                                onUserCreated={[Function]}
                                socialProvider="google"
                                svgIcon={
                                  <SvgMock
                                    fill="#ffffff"
                                    height={20}
                                    width={20}
                                  />
                                }
                              />
                            </View>
                            <View
                              style={
                                Array [
                                  Object {
                                    "flexDirection": "row",
                                    "flexWrap": "wrap",
                                    "justifyContent": "center",
                                    "paddingBottom": 40,
                                    "paddingLeft": 0,
                                    "paddingRight": 0,
                                    "paddingTop": 0,
                                  },
                                ]
                              }
                            >
                              <Text
                                style={
                                  Array [
                                    Object {
                                      "color": "#777",
                                      "lineHeight": 22,
                                    },
                                  ]
                                }
                              >
                                We consider that you agree with 
                              </Text>
                              <Text
                                onPress={[Function]}
                                style={
                                  Array [
                                    Object {
                                      "color": "#1E6EFA",
                                      "lineHeight": 22,
                                      "textDecorationLine": "underline",
                                    },
                                  ]
                                }
                                testID="btn-terms"
                              >
                                Terms for Agreement
                              </Text>
                              <Text
                                style={
                                  Array [
                                    Object {
                                      "color": "#777",
                                      "lineHeight": 22,
                                    },
                                  ]
                                }
                              >
                                 and 
                              </Text>
                              <Text
                                onPress={[Function]}
                                style={
                                  Array [
                                    Object {
                                      "color": "#1E6EFA",
                                      "lineHeight": 22,
                                      "textDecorationLine": "underline",
                                    },
                                  ]
                                }
                                testID="btn-privacy"
                              >
                                Privacy Policy
                              </Text>
                              <Text
                                style={
                                  Array [
                                    Object {
                                      "color": "#777",
                                      "lineHeight": 22,
                                    },
                                  ]
                                }
                              >
                                 by going onto next step.
                              </Text>
                            </View>
                          </View>
                        </View>
                      </RCTScrollView>
                    </RCTSafeAreaView>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      pointerEvents="box-none"
      style={
        Array [
          Object {
            "zIndex": 1,
          },
          Object {
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
        ]
      }
    />
  </RNCSafeAreaProvider>
</View>
`;
